### PR TITLE
Apb 7667 idomatic unit tests

### PR DESCRIPTION
**Pull request by alexander-fisher**

### CODE REVIEW

File: app/uk/gov/hmrc/agentaccesscontrol/controllers/AuthorisationController.scala [hunk 1]
```diff
@@ -36,21 +36,20 @@ class AuthorisationController @Inject()(
...
     val esAuthorisationService: ESAuthorisationService,
-    override val env: Environment,
     cc: ControllerComponents)(implicit val ec: ExecutionContext)
     extends BackendController(cc)
     with AuthAction
     with Logging {
 
   def authorise(authType: String,
-                agentCode: AgentCode,
+                agentCode: String,
```
Comment (CarolineS-QA) [inline app/uk/gov/hmrc/agentaccesscontrol/controllers/AuthorisationController.scala:45]: what's the advantage of converting it in code rather than having the AgentCode type in the app.routes file?

Comment (alexander-fisher) [reply app/uk/gov/hmrc/agentaccesscontrol/controllers/AuthorisationController.scala:45]: Given it's only being used in one place it seemed a bit over-engineered so better for readability to just have the conversion in the code.

File: app/uk/gov/hmrc/agentaccesscontrol/services/AuthorisationService.scala [hunk 2]
```diff
@@ -77,17 +78,12 @@ class AuthorisationService @Inject()(
                       result = false,
                       AccessResponse.toReason(accessResponse))
         Future.successful(accessResponse)
-      case _ =>
```
Comment (alexander-fisher) [inline app/uk/gov/hmrc/agentaccesscontrol/services/AuthorisationService.scala:80]: review whether this case could be hit

Comment (alexander-fisher) [reply app/uk/gov/hmrc/agentaccesscontrol/services/AuthorisationService.scala:80]: authDetails match {
    case nonMtdAgentAuthDetails @ AuthDetails(Some(saAgentReference), _, _, _, _) =>
    case mtdAgentAuthDetails @ AuthDetails(_, Some(arn), _, _, _) =>
    case agentAuthDetails @ AuthDetails(None, _, _, _, _) =>
    case _ =>
}
The 1st and 3rd case cover all potential cases as they test for Some and None on the first item and don't care about anything else so the _ case is unreachable

**Review by CarolineS-QA — APPROVED** [review]
Would be good to get a ticket for some of the TODOs around improving readability / reducing complexity. Kind of out of scope for the integration test passover on this 😅

File: app/uk/gov/hmrc/agentaccesscontrol/controllers/AuthorisationController.scala (hunk 1 above)
Comment (CarolineS-QA) [reply app/uk/gov/hmrc/agentaccesscontrol/controllers/AuthorisationController.scala:45]: fair enough 👍
